File: src/entities/OrderItem.ts
import {
  Entity,
  PrimaryGeneratedColumn,
  Column,
  ManyToOne,
  JoinColumn,
} from "typeorm";
import { Order } from "./Order";
import { MenuItem } from "./MenuItem";
import { decimalTransformer } from "./decimalTransformer";

@Entity("order_items")
export class OrderItem {
  @PrimaryGeneratedColumn("uuid")
  id!: string;

  @Column()
  orderId!: string;

  @ManyToOne(() => Order, (order) => order.items)
  @JoinColumn({ name: "orderId" })
  order!: Order;

  @Column()
  menuItemId!: string;

  @ManyToOne(() => MenuItem, (menuItem) => menuItem.orderItems)
  @JoinColumn({ name: "menuItemId" })
  menuItem!: MenuItem;

  @Column({ type: "int" })
  quantity!: number;

  @Column({ type: "decimal", precision: 10, scale: 2, transformer: decimalTransformer })
  unitPrice!: number;
}

File: flutter_app/lib/core/models/order_item.dart
class OrderItem {
  final String id;
  final String menuItemId;
  final int quantity;
  final double unitPrice;
  final String? menuItemName;

  const OrderItem({
    required this.id,
    required this.menuItemId,
    required this.quantity,
    required this.unitPrice,
    this.menuItemName,
  });

  factory OrderItem.fromJson(Map<String, dynamic> json) => OrderItem(
    id: json['id'] as String,
    menuItemId: json['menuItemId'] as String,
    quantity: json['quantity'] as int,
    unitPrice: (json['unitPrice'] as num).toDouble(),
    menuItemName: json['menuItem']?['name'] as String?,
  );
}
